core/src/main/java/com/linecorp/armeria/internal/common/util/ChannelUtil.java

```diff
@@ -204,38 +210,48 @@ private static boolean canAddChannelOption(@Nullable ChannelOption<?> channelOpt
+                putChannelOption(newChannelOptionsBuilder, epollTcpUserTimeout, tcpUserTimeout);
             } else if (transportType == TransportType.IO_URING &&
                        canAddChannelOption(ioUringTcpUserTimeout, channelOptions)) {
-                newChannelOptionsBuilder.put(ioUringTcpUserTimeout,
-                                             idleTimeoutMillis + TCP_USER_TIMEOUT_BUFFER_MILLIS);
+                putChannelOption(newChannelOptionsBuilder, ioUringTcpUserTimeout, tcpUserTimeout);
             }
         }
 
-        if (pingIntervalMillis > 0 && pingIntervalMillis <= Integer.MAX_VALUE) {
+        if (pingIntervalMillis > 0) {
+            final int intPingIntervalMillis = saturated(pingIntervalMillis);
             if (transportType == TransportType.EPOLL &&
                 canAddChannelOption(epollTcpKeepidle, channelOptions) &&
                 canAddChannelOption(epollTcpKeepintvl, channelOptions) &&
                 canAddChannelOption(ChannelOption.SO_KEEPALIVE, channelOptions)) {
-                newChannelOptionsBuilder.put(ChannelOption.SO_KEEPALIVE, true);
-                newChannelOptionsBuilder.put(epollTcpKeepidle, pingIntervalMillis);
-                newChannelOptionsBuilder.put(epollTcpKeepintvl, pingIntervalMillis);
+                putChannelOption(newChannelOptionsBuilder, ChannelOption.SO_KEEPALIVE, true);
+                putChannelOption(newChannelOptionsBuilder, epollTcpKeepidle, intPingIntervalMillis);
+                putChannelOption(newChannelOptionsBuilder, epollTcpKeepintvl, intPingIntervalMillis);
             } else if (transportType == TransportType.IO_URING &&
                        canAddChannelOption(ioUringTcpKeepidle, channelOptions) &&
                        canAddChannelOption(ioUringTcpKeepintvl, channelOptions) &&
                        canAddChannelOption(ChannelOption.SO_KEEPALIVE, channelOptions)) {
-                newChannelOptionsBuilder.put(ChannelOption.SO_KEEPALIVE, true);
-                newChannelOptionsBuilder.put(ioUringTcpKeepidle, pingIntervalMillis);
-                newChannelOptionsBuilder.put(ioUringTcpKeepintvl, pingIntervalMillis);
+                putChannelOption(newChannelOptionsBuilder, ChannelOption.SO_KEEPALIVE, true);
+                putChannelOption(newChannelOptionsBuilder, ioUringTcpKeepidle, intPingIntervalMillis);
+                putChannelOption(newChannelOptionsBuilder, ioUringTcpKeepintvl, intPingIntervalMillis);
             }
         }
         newChannelOptionsBuilder.putAll(channelOptions);
         return newChannelOptionsBuilder.build();
     }
 
+    private static <T> void putChannelOption(
+            ImmutableMap.Builder<ChannelOption<?>, Object> newChannelOptionsBuilder,
+            ChannelOption<T> channelOption, T value) {
+        newChannelOptionsBuilder.put(channelOption, value);
+    }
+
+    private static int saturated(long value) {
```

Comment:
nit: Could use Guava's `Ints.saturatedCast(long)`?

Reply:
That was what I was looking for. 😅